db/migrate/20170728140134_remove_email_from_enterprises.rb

```diff
@@ -0,0 +1,48 @@
+class RemoveEmailFromEnterprises < ActiveRecord::Migration
+  class Enterprise < ActiveRecord::Base; end
+  class Spree::User < ActiveRecord::Base; end
+
+  class EnterpriseRole < ActiveRecord::Base
+    belongs_to :user, class_name: 'Spree::User'
+    belongs_to :enterprise, class_name: 'Enterprise'
+  end
+
+  def up
+    Enterprise.reset_column_information
+    Spree::User.reset_column_information
+    EnterpriseRole.reset_column_information
```

Comment:
This scares me mainly because I don't understand it. Do you remember what issue this was solving?

Reply:
I remember that there were some migrations that should have defined their own model instead of relying on the current code in `app/models`. When Matt added those, he also added this. I believed it was to avoid ActiveRecord trying to access a column that has been deleted.

Reply:
Michael asked me to redo the migration, with the models defined inside it. It helps stop conflicts when migrations are being run or rolled back after a model has changed. Models are cached when running migrations, so this resets it to make sure the model has access to the current attributes.

http://guides.rubyonrails.org/v3.2/migrations.html#using-models-in-your-migrations